test/util/util.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/util/wait"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1alpha1"
+	v1alpha1servicecatalog "github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset/typed/servicecatalog/v1alpha1"
+)
+
+// WaitForBrokerCondition waits for the status of the named broker to contain
+// a condition whose type and status matches the supplied one.
+func WaitForBrokerCondition(client v1alpha1servicecatalog.ServicecatalogV1alpha1Interface, name string, condition v1alpha1.BrokerCondition) error {
```

Comment:
Would it be useful go make this a bit more generic so that we could use it for other types of conditions, bindings/serviceclasses and instances without having to create things like:
WaitForServiceInstanceCondition, WaitForBindingCondition, etc.
Might be as simple as passing the type of the object we're checking conditions on instead of baking it into the name, I just think having four pretty identical methods (and probably more once we start checking the creation of secrets, etc.) gets a bit unwieldy.